xmtp_mls/src/storage/sql_key_store.rs

```diff
@@ -10,19 +11,22 @@ use super::{
 };
 
 #[derive(Debug)]
+/// CRUD Operations for an [`EncryptedMessageStore`]
 pub struct SqlKeyStore<'a> {
-    store: &'a EncryptedMessageStore,
+    store: Cow<'a, EncryptedMessageStore>,
```

Comment:
I'd love to learn more about what the implications are for wrapping this in a `Cow` are, and what the advantages are. How do I make sure I don't misuse the smart pointer?

Reply:
So long as Cow is always cloning the existing instance, it should be fine since the SQLite connection pool is safe to be cloned. 

My main worry is around anything that causes multiple SQLite pools to exist at the same time pointing to the same DB file. That will cause conflicts and failures on db writes.